File: app/src/main/java/com/nuhkoca/offlineapp/data/local/entity/News.java
package com.nuhkoca.offlineapp.data.local.entity;

import com.google.gson.annotations.SerializedName;
import com.nuhkoca.offlineapp.BR;

import java.util.Date;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "news", indices = @Index(value = "title", unique = true))
public class News extends BaseObservable {

    @PrimaryKey(autoGenerate = true)
    @SerializedName("id")
    private int id;

    @SerializedName("title")
    private String title;

    @SerializedName("author")
    private String author;

    @SerializedName("description")
    private String description;

    @SerializedName("url")
    private String url;

    @SerializedName("urlToImage")
    private String urlToImage;

    @SerializedName("publishedAt")
    private Date publishedAt;

    @SerializedName("content")
    private String content;

    @Bindable
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
        notifyPropertyChanged(BR.id);
    }

    @Bindable
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
        notifyPropertyChanged(BR.title);
    }

    @Bindable
    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
        notifyPropertyChanged(BR.author);
    }

    @Bindable
    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
        notifyPropertyChanged(BR.description);
    }

    @Bindable
    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
        notifyPropertyChanged(BR.url);
    }

    @Bindable
    public String getUrlToImage() {
        return urlToImage;
    }

    public void setUrlToImage(String urlToImage) {
        this.urlToImage = urlToImage;
        notifyPropertyChanged(BR.urlToImage);
    }

    @Bindable
    public Date getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(Date publishedAt) {
        this.publishedAt = publishedAt;
        notifyPropertyChanged(BR.publishedAt);
    }

    @Bindable
    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
        notifyPropertyChanged(BR.content);
    }
}

File: app/src/main/java/com/nuhkoca/offlineapp/data/local/entity/News.kt
package com.nuhkoca.offlineapp.data.local.entity

import com.google.gson.annotations.SerializedName
import com.nuhkoca.offlineapp.BR

import java.util.Date

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "news", indices = [Index(value = ["title"], unique = true)])
class News : BaseObservable() {

    @PrimaryKey(autoGenerate = true)
    @SerializedName("id")
    @get:Bindable
    var id: Int = 0
        set(id) {
            field = id
            notifyPropertyChanged(BR.id)
        }

    @SerializedName("title")
    @get:Bindable
    var title: String? = null
        set(title) {
            field = title
            notifyPropertyChanged(BR.title)
        }

    @SerializedName("author")
    @get:Bindable
    var author: String? = null
        set(author) {
            field = author
            notifyPropertyChanged(BR.author)
        }

    @SerializedName("description")
    @get:Bindable
    var description: String? = null
        set(description) {
            field = description
            notifyPropertyChanged(BR.description)
        }

    @SerializedName("url")
    @get:Bindable
    var url: String? = null
        set(url) {
            field = url
            notifyPropertyChanged(BR.url)
        }

    @SerializedName("urlToImage")
    @get:Bindable
    var urlToImage: String? = null
        set(urlToImage) {
            field = urlToImage
            notifyPropertyChanged(BR.urlToImage)
        }

    @SerializedName("publishedAt")
    @get:Bindable
    var publishedAt: Date? = null
        set(publishedAt) {
            field = publishedAt
            notifyPropertyChanged(BR.publishedAt)
        }

    @SerializedName("content")
    @get:Bindable
    var content: String? = null
        set(content) {
            field = content
            notifyPropertyChanged(BR.content)
        }
}
